### PR TITLE
Fix linking issue in libtorch under macOS

### PR DESCRIPTION
**Pull request by lantiga**
Since yesterday, building libtorch on my macOS results in the following link error:

```
Undefined symbols for architecture x86_64:
  "torch::jit::CppOp::Kind", referenced from:
      torch::jit::CppOp::cloneFrom(torch::jit::Node*) in function.cpp.o
ld: symbol(s) not found for architecture x86_64
```

~~Adding the definition for `torch::jit::CppOp::Kind` and `torch::jit::PythonOp::Kind` in `function.cpp` instead of `ir.cpp` fixes the issue, but I'm not sure it's the best solution.~~

UPDATE: `torch::jit::CppOp::Kind` and `torch::jit::PythonOp::Kind` in `ir.cpp` were inside a `NO_PYTHON` block (facepalm).

### CODE REVIEW

File: torch/csrc/autograd/function.cpp [hunk 1]
```diff
@@ -14,6 +14,14 @@
 #include <utility>
 #include <vector>
 
+namespace torch { namespace jit {
+
+// Sigh, see https://stackoverflow.com/questions/8016780/undefined-reference-to-static-constexpr-char
+constexpr Symbol CppOp::Kind;
+constexpr Symbol PythonOp::Kind;
```
Comment (apaszke) [inline torch/csrc/autograd/function.cpp]: Hmm does it really fix the problem? It shouldn’t matter in which cpp file they are defined I think

**Comment by lantiga** [comment]
That’s what I thought as well. Fact is, moving it fixes the issue for me.

**Comment by lantiga** [comment]
One more quirk: adding it (incorrectly) in both ir.cpp and function.cpp doesn’t lead to a build error on mac, but it does on linux.

**Comment by ezyang** [comment]
That's weird. Can you check your Mac build logs and see if `ir.cpp` is actually being built and linked into libtorch?

**Comment by lantiga** [comment]
That is weird. It took me some time to find that workaround because it wasn't making sense to me.
Here's my link.txt file and build log for master (i.e. without the "fix"): https://gist.github.com/lantiga/f52e4d060ac196c759d82f331c0212a9

**Comment by apaszke** [comment]
Well that makes sense - the definitions are inside a `NO_PYTHON` block!

**Comment by lantiga** [comment]
Oh darn! So sorry, I'll get the fix in in a minute.

**Comment by ezyang** [comment]
Thanks. We definitely should start doing the incremental work of refactoring the code so `NO_PYTHON` blocks are moved into their own directory.

**Comment by lantiga** [comment]
Thank you Edward, and sorry for the silly handling of the issue. It was there in plain sight and it couldn't really have been anything else.

I agree re: refactoring. I'm currently working on #5757, I have capacity to start once that one is done (I'm making progress but I don't have a lot of time this week).

I'll go ahead and open an issue.

**Pull request merged**